tasks/coverage/src/driver.rs

```diff
@@ -150,4 +156,41 @@ impl Driver {
         }
         false
     }
+
+    /// Idempotency test for printing regular expressions.
+    fn check_regular_expressions(&mut self, semantic: &Semantic<'_>) {
+        let allocator = Allocator::default();
+        for literal in semantic.nodes().iter().filter_map(|node| node.kind().as_reg_exp_literal()) {
+            let Some(pattern) = literal.regex.pattern.as_pattern() else {
+                continue;
+            };
+            let printed1 = pattern.to_string();
+            let flags = literal.regex.flags;
+            let printed2 = match PatternParser::new(
+                &allocator,
+                &printed1,
+                ParserOptions {
+                    span_offset: 0,
+                    unicode_mode: flags.contains(RegExpFlags::U) || flags.contains(RegExpFlags::V),
+                    unicode_sets_mode: flags.contains(RegExpFlags::V),
+                },
+            )
+            .parse()
```

Comment:
These code is getting repeated a lot, we should export a top level function `parse_regular_expression(pattern, flags)` from the crate.

Reply:
`oxc_regular_expression::Parser::new(&allocator, format!("/{printed1}/{flags}"), ParserOptions::default())` is still a bit verbose?

I will add it later. 👍🏻

Reply:
This is an unwanted costly memory allocation ;-) format!("/{printed1}/{flags}")